ethcore/blockchain/src/blockchain.rs

```diff
@@ -861,6 +861,25 @@ impl BlockChain {
 		self.block_details.write().clear();
 		self.block_hashes.write().clear();
 		self.block_headers.write().clear();
+		// Fetch best block details
+		let best_block_hash = self.db.key_value().get(db::COL_EXTRA, b"best")
+			.expect("Low-level database error when fetching 'best' block. Some issue with disk?")
```

Comment:
I think it could make sense to move out `writing best_block` to another method because it is not really `clearing the cache`?!

And perhaps return `Err` instead of unwrap/expect

Reply:
> I think that this is sufficient justification for read errors. 

@DemiMarie-parity Did you mean to write "insufficient" here?

Also, there is no writing to disk happening here or elsewhere in the method (the `write()`-calls acquire `RwLocks`) so I'm not sure I understand yours and Niklas' objection.
The pattern of reading from disk and unwrap with the justification you see here is an established pattern in the code base. It is questionable for sure, but not new or relevant to this PR.

Reply:
right, I see. 

It looks `ok` from my side even though I would prefer a method `fn fetch_best_block` instead of having it in `fn clear_cache`